packages/typography/src/Link/Link.spec.tsx

```diff
@@ -141,100 +118,99 @@ describe('packages/typography', () => {
         expect(icon).toBeInTheDocument();
       });
     });
-  });
-
-  describe('when no "href" prop is passed and the "as" prop is not supplied', () => {
-    test('it renders the Link inside of "span" tags', () => {
-      renderLink({});
-      const span = screen.getByText('Link').parentElement;
-      expect(span).not.toBeNull();
-      expect(span!.tagName.toLowerCase()).toBe('span');
-    });
-  });
 
-  describe('when the "as" prop is supplied', () => {
-    test('it renders the correct tag to the DOM', () => {
-      renderLink({
-        href: 'http://localhost:9001',
-        /* @ts-expect-error to show that "as" overrides "href" */
-        as: 'div',
+    describe('inferred polymorphic behavior', () => {
+      test('when the "as" prop is supplied its value is respected', () => {
+        renderLink({
+          href: 'http://localhost:9001',
+          /* @ts-expect-error to show that "as" overrides "href" */
+          as: 'div',
+        });
+        const div = screen.getByText('Link').parentElement;
+        expect(div).toBeVisible();
+        expect(div!.tagName.toLowerCase()).toBe('div');
       });
-      const div = screen.getByText('Link').parentElement;
-      expect(div).toBeVisible();
-      expect(div!.tagName.toLowerCase()).toBe('div');
-    });
-  });
 
-  describe('anchor props', () => {
-    test('href is always passed to rendered component when the prop is supplied', () => {
-      renderLink({ href: 'test' });
-      const a = screen.getByText('Link').parentElement;
-      expect(a).toHaveAttribute('href');
-      expect(a?.getAttribute('href')).toBe('test');
-    });
+      test('when an href prop is supplied, its rendered as an anchor', () => {
+        renderLink({ href: 'string' });
+        const anchor = screen.getByText('Link').parentElement;
+        expect(anchor).toBeVisible();
+        expect(anchor!.tagName.toLowerCase()).toBe('a');
+      });
 
-    test('it respects "target" and "rel" values when target is set', () => {
-      renderLink({ href: 'string', target: 'test' });
-      const a = screen.getByText('Link').parentElement;
-      expect(a).toHaveAttribute('target');
-      expect(a).not.toHaveAttribute('rel');
-      expect(a?.getAttribute('target')).toBe('test');
+      test('by default it renders as a span', () => {
```

Comment:
Nit: the test name is a bit confusing. I'd maybe change it to "renders as a span when no `as` or `href` is provided"

Reply:
👍